test/testdata/goldens/help_test/pub add --help.txt

```diff
@@ -29,12 +31,14 @@ For example:
     `dart pub add 'foo:{"sdk":"flutter"}'`
   * Add a git dependency:
     `dart pub add 'foo:{"git":"https://github.com/foo/foo"}'`
+  * Add a dependency override:
+    `dart pub add 'override:foo:1.0.0`
   * Add a git dependency with a path and ref specified:
     `dart pub add \
       'foo:{"git":{"url":"../foo.git","ref":"branch","path":"subdir"}}'`
 
-Usage: pub add [options] [dev:]<package>[:descriptor] [[dev:]<package>[:descriptor]
-       ...]
+Usage: pub add [options] [dev:|override:]<package>[:descriptor]
+       [dev:|override:]<package>[:descriptor] ...]
```

Comment:
We could also consider
```
Usage: pub add [options] [dev|override:]<package>[:descriptor]
       [dev|override:]<package>[:descriptor] ...]
```
or
```
Usage: pub add [options] [<dev|override>:]<package>[:descriptor]
       [<dev|override>:]<package>[:descriptor] ...]
```

or 
```
Usage: pub add [options] [<section>:]<package>[:descriptor]
       [<section>:]<package>[:descriptor] ...]
```

Or maybe, we omit this from the usage line, and explain it further down.
We might just be confusing usings.